apriltag/src/main/native/cpp/jni/AprilTagJNI.cpp

```diff
@@ -609,25 +597,38 @@ Java_edu_wpi_first_apriltag_jni_AprilTagJNI_estimatePose
 /*
  * Class:     edu_wpi_first_apriltag_jni_AprilTagJNI
  * Method:    generate16h5AprilTagImage
- * Signature: (I)Ljava/lang/Object;
+ * Signature: (Ljava/lang/Object;JI)V
  */
-JNIEXPORT jobject JNICALL
+JNIEXPORT void JNICALL
 Java_edu_wpi_first_apriltag_jni_AprilTagJNI_generate16h5AprilTagImage
-  (JNIEnv* env, jclass, jint id)
+  (JNIEnv* env, jclass, jobject frameObj, jlong framePtr, jint id)
 {
-  return MakeJObject(env, AprilTag::Generate16h5AprilTagImage(id));
+  auto* frame = reinterpret_cast<wpi::RawFrame*>(framePtr);
+  if (!frame) {
+    nullPointerEx.Throw(env, "frame is null");
+    return;
+  }
+  bool newData = AprilTag::Generate16h5AprilTagImage(frame, id);
+  wpi::SetFrameData(env, rawFrameCls, frameObj, *frame, newData);
 }
 
 /*
  * Class:     edu_wpi_first_apriltag_jni_AprilTagJNI
  * Method:    generate36h11AprilTagImage
- * Signature: (I)Ljava/lang/Object;
+ * Signature: (Ljava/lang/Object;JI)V
  */
-JNIEXPORT jobject JNICALL
+JNIEXPORT void JNICALL
 Java_edu_wpi_first_apriltag_jni_AprilTagJNI_generate36h11AprilTagImage
-  (JNIEnv* env, jclass, jint id)
+  (JNIEnv* env, jclass, jobject frameObj, jlong framePtr, jint id)
```

Comment:
Wait, we pass in a RawFrame object but don't use it?

Reply:
It does get used.

Reply:
Average code review on cell phone L